From 6745709b051e62beec827dab7533d0b69bd54c9a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micka=C3=ABl=20Sala=C3=BCn?= <mic@digikod.net>
Date: Wed, 21 Jun 2023 13:55:13 +0200
Subject: [PATCH 1/4] WORKSHOP: Replace old libcl with new ocl-icd

---
 PKGBUILD | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/PKGBUILD b/PKGBUILD
index 2441f4ae7253..1d4eb7f920c8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,7 +9,7 @@ url="http://www.imagemagick.org/"
 license=('custom')
 makedepends=('libltdl' 'lcms2' 'libxt' 'fontconfig' 'libxext' 'ghostscript'
              'openexr' 'libwmf' 'librsvg' 'libxml2' 'liblqr' 'openjpeg2'
-             'opencl-headers' 'libcl' 'libwebp' 'subversion' 'glu')
+             'opencl-headers' 'ocl-icd' 'libwebp' 'subversion' 'glu')
 source=(http://www.imagemagick.org/download/ImageMagick-${pkgver%.*}-${pkgver##*.}.tar.xz{,.asc}
         perlmagick.rpath.patch)
 sha1sums=('ab3a16e1ac223153a76c98d20ed6dda98116d91a'
@@ -45,7 +45,7 @@ check() {
 
 package_imagemagick() {
   pkgdesc="An image viewing/manipulation program"
-  depends=('libltdl' 'lcms2' 'libxt' 'fontconfig' 'libxext' 'liblqr' 'libcl')
+  depends=('libltdl' 'lcms2' 'libxt' 'fontconfig' 'libxext' 'liblqr' 'ocl-icd')
   optdepends=('imagemagick-doc: for additional information'
               'ghostscript: for Ghostscript support' 
               'openexr: for OpenEXR support' 
-- 
2.43.0

